docs/specification/draft/basic/authorization.mdx

```diff
@@ -317,70 +270,10 @@ Servers **MUST** return appropriate HTTP status codes for authorization errors:
 ### 2.9 Implementation Requirements
 
 1. Implementations **MUST** follow OAuth 2.1 security best practices
-2. PKCE is **REQUIRED** for all clients
-3. Token rotation **SHOULD** be implemented for enhanced security
-4. Token lifetimes **SHOULD** be limited based on security requirements
-
-### 2.10 Third-Party Authorization Flow
-
-#### 2.10.1 Overview
-
-MCP servers **MAY** support delegated authorization through third-party authorization
-servers. In this flow, the MCP server acts as both an OAuth client (to the third-party
-auth server) and an OAuth authorization server (to the MCP client).
-
-#### 2.10.2 Flow Description
-
-The third-party authorization flow comprises these steps:
-
-1. MCP client initiates standard OAuth flow with MCP server
-2. MCP server redirects user to third-party authorization server
-3. User authorizes with third-party server
-4. Third-party server redirects back to MCP server with authorization code
-5. MCP server exchanges code for third-party access token
-6. MCP server generates its own access token bound to the third-party session
-7. MCP server completes original OAuth flow with MCP client
-
-```mermaid
-sequenceDiagram
-    participant B as User-Agent (Browser)
-    participant C as MCP Client
-    participant M as MCP Server
-    participant T as Third-Party Auth Server
-
-    C->>M: Initial OAuth Request
-    M->>B: Redirect to Third-Party /authorize
-    B->>T: Authorization Request
-    Note over T: User authorizes
-    T->>B: Redirect to MCP Server callback
-    B->>M: Authorization code
-    M->>T: Exchange code for token
-    T->>M: Third-party access token
-    Note over M: Generate bound MCP token
-    M->>B: Redirect to MCP Client callback
-    B->>C: MCP authorization code
-    C->>M: Exchange code for token
-    M->>C: MCP access token
-```
-
-#### 2.10.3 Session Binding Requirements
-
-MCP servers implementing third-party authorization **MUST**:
-
-1. Maintain secure mapping between third-party tokens and issued MCP tokens
-2. Validate third-party token status before honoring MCP tokens
-3. Implement appropriate token lifecycle management
-4. Handle third-party token expiration and renewal
-
-#### 2.10.4 Security Considerations
-
-When implementing third-party authorization, servers **MUST**:
-
-1. Validate all redirect URIs
-2. Securely store third-party credentials
-3. Implement appropriate session timeout handling
-4. Consider security implications of token chaining
-5. Implement proper error handling for third-party auth failures
+1. PKCE is **REQUIRED** for all MCP clients
+1. MCP servers that also act as an AS:
+    1. **SHOULD** implement token rotation for enhanced security
```

Comment:
Why is this a requirement unique to a tightly coupled AS/RS?

Reply:
@aaronpk if you're using a third-party AS, you would have no control over token rotation policies, no? That is, until you actually have to refresh your own cached tokens.

Reply:
ok we'll come back to this in a follow-up PR. I'd rather take out this recommendation entirely since the recommendation is really to follow OAuth 2.1's requirements on refresh token rotation.